lib/slimmer/processors/feedback_url_swapper.rb

```diff
@@ -0,0 +1,34 @@
+module Slimmer::Processors
+  class FeedbackURLSwapper
+    def initialize(request, headers)
+      @headers = headers
+      @request = request
+    end
+
+    def filter(_src, dest)
+      return dest unless is_gem_layout?
+
+      original_url_without_pii = remove_pii(@request.base_url + @request.fullpath)
+      dest.at_css(".gem-c-feedback input[name='url']").set_attribute("value", original_url_without_pii)
+
+      full_path_without_pii = remove_pii(@request.fullpath)
+      dest.at_css(".gem-c-feedback input[name='email_survey_signup[survey_source]']").set_attribute("value", full_path_without_pii)
+
+      dest
+    end
+
+  private
+
+    # This PII removal is also found in the [feedback component in the GOV.UK
+    # Publishing Components gem](https://git.io/JcCIE), and any changes made
+    # need to be kept in sync.
+    def remove_pii(string)
+      email_regex = /[^\s=\/?&]+(?:@|%40)[^\s=\/?&]+/
+      string.force_encoding("UTF-8").gsub(email_regex, "[email]")
```

Comment:
```suggestion
      string.dup.force_encoding("UTF-8").gsub(email_regex, "[email]")
```

This should fix the test failure you've got. 

The reason it's failing is because `force_encoding` modifies the string that's passed in rather than return a new string, which means it's trying to change the `fullpath` attribute of the request. That's set a frozen string in that class which means it bombs when we try to modify it (it's quite nice they have that set-up otherwise we could be accidentally borking the request object).

The `force_encoding` method modifies the source string, which

Reply:
I've put the `gsub` first, which I think works because it returns a copy of the string which isn't frozen:

```
 string.gsub(email_regex, "[email]").force_encoding("UTF-8")
```

but I like using `dup` as it's clearer that the string is being copied and then edited. 

Oddly, this didn't fail locally.

Reply:
Super, thanks for that.

It did fail for me locally. My best guess as to why you didn't experience a fail is because, as a gem, this doesn't have a Gemfile.lock commited and we specify a dependency of any version of Rack. So locally you could have an older Gemfile.lock on your local machine that means this is using a version of Rack before they switched to frozen attributes. 

You may well find that running `bundle update` allows you to replicate the failure.

FYI the version of Rack for me is:

```
➜  slimmer git:(add-feedback-url-swapper) ✗ bundle info rack
  * rack (2.2.3)
	Summary: A modular Ruby webserver interface.
	Homepage: https://github.com/rack/rack
	Path: /Users/kevindew/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rack-2.2.3
```

Reply:
Yes, and I appear to be using a different version of Ruby as well 🤯 - but I can replicate the error now. Cheers!